GliaWidgets/View/Transition.swift

```diff
@@ -0,0 +1,95 @@
+import UIKit
+
+class Transition: NSObject {
+    var circle = UIView()
+    var startingPoint = CGPoint.zero {
+        didSet {
+            circle.center = startingPoint
+        }
+    }
+    var duration = 0.4
+
+    enum TransitionMode: Int {
+        case present, dismiss, pop
+    }
+
+    var transitionMode: TransitionMode = .present
+}
+
+// MARK: - UIViewControllerAnimatedTransitioning
+
+extension Transition: UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return duration
+    }
+    
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        let containerView = transitionContext.containerView
+        
+        if transitionMode == .present {
+            if let presentedView = transitionContext.view(forKey: UITransitionContextViewKey.to) {
```

Comment:
So currently this function is responsible for both presenting and dismissing the view, which makes it kinda bulky. I suggest splitting it in two, basically creating `presentView(...)` and `dismissView(...)` private methods and making the only responsibility of `animateTransition(...)` to choose which one to call depending on `transitionMode`. You can pass `transitionContext` parameter along to them, of course.

I feel like that would go in line with single-responsibility principle of SOLID.

As a bonus, you would get rid of huge chunks of code being indented to the right from being thrown into an if/else statement :)